src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java

```diff
@@ -142,4 +147,71 @@ public boolean isElementVisible(WebElement element) {
   public void openHome(String wikiURL) {
     getUrl(wikiURL);
   }
+
+  /**
+   * Make sure element is ready to be clicked and click on it The separation of this method has
+   * particular reason. It allows global modification of such click usages. This way it is very easy
+   * to control what criteria have to be met in order to click on element
+   *
+   * @param element to be clicked on
+   */
+  protected void waitAndClick(WebElement element) {
+    wait.forElementClickable(element).click();
+  }
+
+  protected void waitAndSendKeys(WebElement element, String keys) {
+    wait.forElementVisible(element);
+    element.sendKeys(keys);
+  }
+
+  protected void verifyTextInElement(WebElement element, String text) {
+    wait.forElementVisible(element);
+    Assertion.assertEquals(element.getText(), text);
+  }
+
+  /**
+   * Verify if element inside element of the provided list has given text. This method assumes list
+   * element is not the final target of verification This method assumes list element is the parent
+   * of target element *
+   *
+   * @param list           List that contains the parent element
+   * @param elementLocator Locator of target element, that is child of its parent element
+   * @param text           Text to be compared
+   */
+  protected void verifyTextInListElements(List<WebElement> list, By elementLocator, String text) {
+    for (WebElement elem : list) {
+      verifyTextInElement(elem.findElement(elementLocator), text);
+    }
+  }
+
+  /**
+   * Verify if element of the provided list has given text.
+   *
+   * @param list List that contains the element
+   * @param text Text to be compared
+   */
+  protected void verifyTextInListElements(List<WebElement> list, String text) {
+    for (WebElement elem : list) {
+      verifyTextInElement(elem, text);
+    }
+  }
+
+  /**
+   * Verify if element of the provided list has given text.
+   *
+   * @param list List that contains the element
+   * @param text Text to be compared
+   */
+  protected void clickByListElementText(List<WebElement> list, String text) {
+    for (WebElement elem : list) {
+      wait.forElementVisible(elem);
+      if (elem.getText().equals(text)) {
+        waitAndClick(elem);
+        return;
+      }
+    }
+    throw new RuntimeException("element with text " + text + "not found in the list");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)